src/main/java/com/artipie/rpm/RpmMetadata.java

```diff
@@ -59,9 +67,27 @@ public Remove(final MetadataItem... items) {
         /**
          * Removes records from metadata by RPMs checksums.
          * @param checksums Rpms checksums  to remove by
+         * @throws IOException On io-operation result error
          */
-        public void perform(final Collection<String> checksums) {
-            throw new NotImplementedException("Will be implemented later");
+        public void perform(final Collection<String> checksums) throws IOException {
+            for (final MetadataItem item : this.items) {
+                final XmlMaid maid;
+                final Path temp = Files.createTempFile("rpm-index", "xml");
+                final long res;
+                try (OutputStream out = new BufferedOutputStream(Files.newOutputStream(temp))) {
+                    if (item.type == XmlPackage.PRIMARY) {
+                        maid = new XmlPrimaryMaid.Stream(item.input, out);
+                    } else {
+                        maid = new XmlMaid.ByPkgidAttr.Stream(item.input, out);
+                    }
+                    res = maid.clean(checksums);
+                }
+                try (InputStream input = new BufferedInputStream(Files.newInputStream(temp))) {
+                    new XmlAlter.Stream(input, item.out)
+                        .pkgAttr(item.type.tag(), String.valueOf(res));
+                }
+                Files.delete(temp);
```

Comment:
@olenagerasimova this line is potential cause of resource leaks, so it should be placed in `finally` block